marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/Database/PostgreSQL.hs

```diff
@@ -792,11 +814,11 @@ getUTxOs = GetUTxOs $ fmap Api.UTxOs . \case
              , asset.name :: bytea?
              , assetOut.quantity :: bigint?
           FROM chain.txOut         AS txOut
+          JOIN addresses           AS addr     ON addr.address = txOut.address AND CAST(MD5(addr.address) AS uuid) = CAST(MD5(txOut.address) AS uuid)
```

Comment:
Out of curiosity - do this `JOIN` fail if we skip direct address equality check `addr.address = txOut.address`?

Reply:
only if there is an MD5 hash collision. It's not a significant amount of overhead to add this recheck condition to the filter, the important thing is that it does a bitmap heap scan instead of a seq scan to produce the initial result set.